clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -2950,12 +2950,11 @@ static void handleSectionAttr(Sema &S, Decl *D, const ParsedAttr &AL) {
   }
 }
 
-static bool isValidCodeModelAttr(Sema &S, StringRef Str) {
-  if (S.Context.getTargetInfo().getTriple().isLoongArch()) {
+static bool isValidCodeModelAttr(llvm::Triple Triple, StringRef Str) {
```

Comment:
Triple contains a `std::string`, I wouldn't treat it as a cheap-to-copy value type:
https://github.com/llvm/llvm-project/blob/main/llvm/include/llvm/TargetParser/Triple.h#L321

Reply:
Right. This should be fixed.

Reply:
done